pandas/io/iceberg.py

```diff
@@ -0,0 +1,89 @@
+from typing import (
+    Any,
+)
+
+from pandas.compat._optional import import_optional_dependency
+
+from pandas import DataFrame
+
+
+def read_iceberg(
+    table_identifier: str,
+    catalog_name: str | None = None,
+    catalog_properties: dict[str, Any] | None = None,
+    row_filter: str | None = None,
+    selected_fields: tuple[str] | None = None,
+    case_sensitive: bool = True,
+    snapshot_id: int | None = None,
+    limit: int | None = None,
+    scan_properties: dict[str, Any] | None = None,
+) -> DataFrame:
+    """
+    Read an Apache Iceberg table into a pandas DataFrame.
```

Comment:
Might be good to flag as `experimental` so if we revisit and implement the IO plugin model we can pivot this to that model a little quicker.

Reply:
Makes sense, thanks for all the feedback.